### PR TITLE
update readme formatting

### PR DESCRIPTION
**Pull request by shayanhabibi**
Some formatting suggestions to make the readme cleaner and less choc full of text

### CODE REVIEW

**Comment by haxscramper** [comment]
I don't think it is necessary to "Directions", "Near-term development" portions of the readme in spoilers, they are important enough to keep them open.

**Comment by shayanhabibi** [comment]
I would disagree; I think that having graphics of the near-term development instantly viewable is a good idea, but the directions should be slimmed into a short stunner that contains the meat of the project and then the rest hidden within a spoiler.

Until that happens I think it should be kept in a spoiler (as its a huge block of text)

The idea is to create a cleaner and easier to navigate readme.

**Comment by shayanhabibi** [comment]
Actually I would say there are a couple short stunners already in the direction, I just don't know which one saem would want to be forerunner

**Comment by saem** [comment]
> Actually I would say there are a couple short stunners already in the direction, I just don't know which one saem would want to be forerunner

Curse of knowledge, I don't either.

The main thing this needs to do is:
- show new folks the what, how, and why of contribution
- remember things for current contributors so they can refer back

Practically speaking, @shayanhabibi I think you're representative of a major part of the audience. Eager and able, now need to understand enough to turn that into action.

**Comment by haxscramper** [comment]
Maybe some sort of FAQ? 

From discussion in `#main` - I guess these questions worth mentioning/answering. Probably something else as well, but I just went over and noted questions that seemed meaningful enough


-------------

> what's the rational?

We got tired of the way Nim is going, especially development process
> What are you going to do now?

https://github.com/nim-works/nimskull/projects
> Will this break my code?

Maybe https://github.com/nim-works/nimskull/issues/8 https://github.com/nim-works/nimskull#direction

> Will you plan to merge changes back?

Generally speaking no, but there might be some exceptions
> Can I help somehow?

Presently we're more interested in people contributing than growing a
community, so if you want to see it sooner, get on helping hax with language spec tests
> Any chat room on matrix/irc/discord?

Not now, we will add it sometimes later
> So the aim isn't just for a different compiler, but a new language? Similar
> but different? So tutorials / books will be written eventually?

Just more well-defined spec through excessive testing, which might lead to slight
language changes to make it more correct. At least that's my take.

---------------

**Comment by haxscramper** [comment]
In addition to that (elaborate on the "direction where community is going")

> failing commit rates, lack of future vision from the community leadership, half a dozen different directions the language is trying to evolve while having only a couple of fully-time contributors, lack of transparency when it comes to discussion of the new features, complete unwillingness to break anything even if it means sinking deeper and deeper in backwards compat hacks while having very few core contributors, dependence on status basically tying araq's hands and shoving their targeted features as they see fit
> we added 15 new stdlib modules
> 2021 was supposed to be the year of tooling
> do you know what is the 1.8 target features?
> do you know what is the main target objective right now?
> do you know the roadmap for the commuity in the next quarter at least?
> It is not that I don't I feel bad about the language, not in the slightest (well, maybe there are some stuff in the stdlib that I don't like, but that is not as fundamental),
> I like the language and where it is heading
> despite all of these shortcomings
> but man it is starting harder to figure out where are we heading

^ (from discussion in  `#offtopic, I did rephrase some things to avoid directly naming anyone)

> if you mean whether we are willing to do breakages - then yes, but we are not aiming to just destroy everything for the sake of some abstract understanding of what constitutes better
> But in the end right now this is not something that we are really inviting people to contribute and discuss, mostly because there are a lot of things that need to be figured out
> So we are afraid of taking too much different options at once, most of which we won't be able to meaningfully process

^ That one is for telling people about our plan for managing community

**Comment by shayanhabibi** [comment]
Dunno if I want to include too much of the negatives from that last post. IMO the read-me should instill an air of positivity around this project; I'd rather avoid the rest and leave that for perhaps a separate doc that can be linked from the readme..

Anyway, have a looksy @haxscramper and see what you think.

File: readme.md [hunk 1]
```diff
@@ -1,191 +1,303 @@
...
+</details>
+
+<details>
+<summary class="blue">Will you plan to merge Nim upstream changes?</summary>
+Generally? No. There will likely be exceptions to this. Our current focus remains on solidifying the foundations before moving forward.
+</details>
+
+<details>
+<summary class="blue">Can I help somehow?</summary>
+Presently we're very interested in people contributing than to growing our community; so if you would like to contribute, a good start is to help haxscramper with the language spec tests
```
Comment (haxscramper) [inline readme.md]: > a good start is to help haxscramper with the language spec tests 

Don't redirect to me, redirect to #10 for to-do, or #28 for examples on how this should be done.

**Comment by haxscramper** [comment]
Aside from the "help haxscramper" (I'm just a contributor here, people should be thinking about helping the project itself), I think it looks good.

I added in negative parts as well mostly for the sake of reference, it does not have to be included directly in the README (though "we believe that an element of the vision was lost" can be elaborated a little more)

**Comment by shayanhabibi** [comment]
Hahaha yeah I probably should have looked at the convo on matrix before copy pasting the FAQ answers. I thought you were actually suggesting those as answers

**Comment by saem** [comment]
Are we g2g in this?

**Comment by haxscramper** [comment]
>  > a good start is to help haxscramper with the language spec tests 

> Don't redirect to me, redirect to #10 for to-do, or #28 for examples on how this should be done.

https://github.com/nim-works/nimskull/pull/32#discussion_r740924218 still not fixed, otherwise it looks good to me

File: readme.md (hunk 1 above)
Comment (saem) [inline readme.md]:
```suggestion
Presently we're very interested in people contributing than to growing our community; so if you would like to contribute, a good start is to help the [language spec as tests effort which is being led be @haxscramper](https://github.com/nim-works/nimskull/issues/10). If you're willing to drive deeper into the compiler then see the [nkError refactor to make the compiler approachable](https://github.com/nim-works/nimskull/projects) project.
```

**Comment by saem** [comment]
> > > a good start is to help haxscramper with the language spec tests
> 
> > Don't redirect to me, redirect to #10 for to-do, or #28 for examples on how this should be done.
> 
> [#32 (comment)](https://github.com/nim-works/nimskull/pull/32#discussion_r740924218) still not fixed, otherwise it looks good to me

I used the suggest feature, how does the suggestion look?

**Comment by haxscramper** [comment]
good, I think we can leave it at that

File: readme.md (hunk 1 above)
Comment (saem) [reply readme.md]: I'm on my phone so going to shy away from hitting commit as I can't really ensure the formatting and everything checkout.

So @shayanhabibi can you haz a look and accept if acceptable or adjust as you'd like?

**Pull request merged**